src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -2687,6 +2687,49 @@ LogicalResult ONNXConstantOp::inferShapes(
 // Concat
 //===----------------------------------------------------------------------===//
 
+static LogicalResult verify(ONNXConcatOp op){
+  ONNXConcatOpAdaptor operandAdaptor =
+    ONNXConcatOpAdaptor(op);
+
+  int inputNum = op.getNumOperands();
+  // Check all inputs.
+  for (int i = 0; i < inputNum; ++i) {
+    if (!hasShapeAndRank(operandAdaptor.getOperands()[i])) {
+      // Won't be able to do any checking at this stage.
+      return success();
+    }
+  }
+  // Checking value of axis parameter.
+  auto commonType = operandAdaptor.getOperands()[0].getType().cast<RankedTensorType>();
+  auto commonShape = commonType.getShape();
```

Comment:
I personally prefer to not use `auto` where is not immediately clear what is the type of the variable.  In this case what is the type returned by getShape() ?  Use that type to declare `commonShape`. This is also LLVM policy (see https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable)

P.S.: on line 2703 is fine to use `auto` because is clear that the type is `RankedTensorType` given there is a cast.